File: test/model/TeamSearchResultMatcher.scala
package model

import util.CompositeMatcher
import org.hamcrest.{TypeSafeMatcher, Description}

object TeamSearchResultMatcher {
    def aTeamSearchResult = new TeamSearchResultMatcher()
}

class TeamSearchResultMatcher extends CompositeMatcher[TeamSearchResult] {
    def named(name: String): TeamSearchResultMatcher = {
        add(new TypeSafeMatcher[TeamSearchResult] () {
            def matchesSafely(team: TeamSearchResult): Boolean = team.name == name

            def describeTo(description: Description) {
                description.appendText(s"named $name")
            }
        })
        this
    }
}

File: src/test/java/model/TeamSearchResultMatcher.java
package model;

import org.hamcrest.Description;
import org.hamcrest.TypeSafeMatcher;
import util.CompositeMatcher;

public class TeamSearchResultMatcher extends CompositeMatcher<TeamSearchResult> {
    private TeamSearchResultMatcher() {
        super();
    }

    public static TeamSearchResultMatcher aTeamSearchResult() {
        return new TeamSearchResultMatcher();
    }

    public TeamSearchResultMatcher named(final String name) {
        add(new TypeSafeMatcher<TeamSearchResult>() {
            @Override
            public boolean matchesSafely(TeamSearchResult target) {
                return target.name.equals(name);
            }
            @Override
            public void describeTo(Description description) {
                description.appendText(String.format("whose name is %s", name));
            }
        });
        return this;
    }
}
